minio/minio
# Helm: add cilium networkpolicy

## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

his PR adds a `ciliumnetworkpolicy` that is equivalent to the standard `networkpolicy` already present in the templates. As cilium usage as a CNI is rising and the usage of ciliumnetworkpolicies is more and more widespread, having the possibility to deploy those directly from a setting in the values would be a time-saving option for a lot of deployments.

As this PR adds a new condition for the already existing `networkpolicy`, this is a breaking change for those using the chart with `networkpolicies` enabled as they'll have to add the new field to their values file.

## Motivation and Context

As I tried to use `minio` alongside `loki` on a cluster which was using `cilium` as its CNI, the minio pod was not able to work properly. Especially the job supposed to interact with the main minio pod was not able to complete so I had to manually create a cilium netpol.

## How to test this PR?

One may ltemplate the chart or even deploy it to a cluster with `cilium` as the CNI to make sure the pod is running as expected and the job is able to complete.

## Types of changes

## Checklist:
